src/ip.rs

```diff
@@ -1,72 +1,161 @@
 use std::net::{IpAddr, Ipv4Addr, Ipv6Addr};
 
-/// The kinds of networks that can be simulated in turmoil
-#[derive(Debug, Clone, PartialEq, Eq, Default)]
-pub enum IpVersion {
-    /// An Ipv4 network with an address space of 192.168.0.0/16
-    #[default]
-    V4,
-    /// An local area Ipv6 network with an address space of fe80::/64
-    V6,
+/// The address layout of the underlying network.
+///
+/// The default value is an Ipv4 subnet with addresses
+/// in the range `192.168.0.0/16`.
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+pub enum IpNetwork {
+    /// An Ipv4 capable network, with a given subnet address range.
+    V4(Ipv4Network),
+    /// An Ipv6 capable network, with a given subnet address range.
+    V6(Ipv6Network),
 }
```

Comment:
I think this is a reasonable, pragmatic change, but it doesn't seem like this is what we really want to end up with. For example, what if I want to have a setup where a single machine has an admin process on it that listens with v4 and creates N processes (in response to API calls) with v6 addresses. I think this is the sort of setup you're aiming to support (is that right?), and so I think there is a design smell here.

I'm not suggesting you change this for this review (again, your change is pragmatic), but we should think about how this plays out and be mindful it's likely to cause maintenance problems.